### PR TITLE
:art: feat {Application Design }User menu Ref  #8

### PR DESCRIPTION
**Pull request by trevin-livele**
# Description

The changes in the issue #8  regards the user menu.I have updated the ui and made it meet the design needs and also separated the logout functionality to a separate component.

Fixes # (issue #8 ) 

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Screenshot 
usermenu

# demo

https://user-images.githubusercontent.com/60538411/190017317-4cd20591-ddeb-4b53-b301-5f96ce84c1a9.mp4

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/elements/layout/user-menu/src/lib/components/app-logout/app-logout.component.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<button mat-menu-item (click)="logout()">Log out</button>
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/components/app-logout/app-logout.component.html:1]: Use multi-language

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+      <span class="material-icons" id="dropdown-icon">arrow_drop_down</span>
+    </div>
+    <div class="role">
+      Admin&nbsp;
+    </div>
   </button>
-
-  <mat-menu #userMenu="matMenu" xPosition="before">
+  <mat-menu #userMenu="matMenu">
     <button mat-menu-item>My profile</button>
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html:13]: Use multilanguage

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
-    {{ user.photoUrl }}
-    <i class="far fa-user-circle user-image"></i> {{ user.displayName }}&nbsp;<!-- -&nbsp;{{user.phone}}&nbsp; -->
-    <i class="fas fa-caret-down"></i>
+    <div class="items">
+      <img src="{{ user.photoUrl }}" alt="" class="user-img">
+      <span class="username">{{ user.displayName }}.&nbsp;</span>
+      <span class="material-icons" id="dropdown-icon">arrow_drop_down</span>
+    </div>
+    <div class="role">
+      Admin&nbsp;
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html:9]: Use multi-language

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss [hunk 4]
```diff
@@ -1,5 +1,48 @@
...
+
+.outer {
+  height: 49px;
+  width: 210px;
+  background-color: white;
+  border-radius: 5px;
+}
+
+.outer:hover {
+  background-color: #fafafa;
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss:22]: Implement using scss variables for all colors. Variables to use are in libs/elements/theming/_variables.scss
Example:
`background-color: var(--variable-name)`

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss [hunk 5]
```diff
@@ -1,5 +1,48 @@
...
+.items {
+  display: flex;
+  flex-direction: row;
+}
+
+.role {
+  margin-top: -1.5pc;
+  font-size: small;
+  margin-left: -1.7pc;
+color: #8F8F8F;
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss:41]: Same here. Use scss variables

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss [hunk 6]
```diff
@@ -1,5 +1,48 @@
...
+  border-radius: 10px;
+  height: 35px;
+  width: 35px;
+  margin: 5px;
+}
+
+.outer {
+  height: 49px;
+  width: 210px;
+  background-color: white;
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss:17]: Use scss variables. See next comment

**Review by ReaganCn — COMMENTED** [review]
Please have a look at the code changes requested. In regards to the User menu component, you can fit only at the top menu bar, because now it's between the top menu bar and the breadcrumb.

Otherwise, keep it up!

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Please have a look at the code changes requested. In regards to the User menu component, you can fit only at the top menu bar, because now it's between the top menu bar and the breadcrumb.

Otherwise, keep it up!

**Comment by trevin-livele** [comment]
hello sir @ReaganCn i have made the changes

**Review by ReaganCn — APPROVED** [review]
Good work Trevin! 💪

**Comment by trevin-livele** [comment]
> Good work Trevin! muscle

:+1: thank you sir

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html (hunk 3 above)
Comment (trevin-livele) [reply libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html]: hello sir @jrosseel  I have changed the roles area sir its now dynamic.For the arrow_drop_down it comes from mat icons  i have to call it that way {arrow_drop_down }  if i remove it the drop down icon disappears.

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.ts [hunk 7]
```diff
@@ -1,31 +1,23 @@
 import { Observable } from 'rxjs';
-
 import { Component } from '@angular/core';
 import { Router } from '@angular/router';
-
 import { User } from '@iote/bricks';
-import { UserService, AuthService} from '@ngfi/angular';
+import { UserService } from '@ngfi/angular';
+import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.ts:6]: We are now using the cdn for implementing font awesome icons. See: https://github.com/italanta/elewa/pull/97

File: libs/elements/layout/user-menu/src/lib/user-menu.module.ts [hunk 8]
```diff
@@ -2,21 +2,32 @@ import { NgModule } from '@angular/core';
...
-            MaterialDesignModule, MaterialBricksModule, FlexLayoutModule,
-            ],
+  imports: [
+    CommonModule,
+    RouterModule,
+    MultiLangModule,
+    MaterialDesignModule,
+    MaterialBricksModule,
+    FlexLayoutModule,
+    FontAwesomeModule
```
Comment (ReaganCn) [inline libs/elements/layout/user-menu/src/lib/user-menu.module.ts:25]: Remove Font Awesome Module import. See PR https://github.com/italanta/elewa/pull/97

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Good work. Check the below changes. We changed the way we implement icons

File: libs/elements/layout/user-menu/src/lib/components/app-logout/app-logout.component.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { Component } from '@angular/core';
+import { UserService, AuthService} from '@ngfi/angular';
+import { User } from '@iote/bricks';
+import { Observable } from 'rxjs';
```
Comment (IanOdhiambo9) [inline libs/elements/layout/user-menu/src/lib/components/app-logout/app-logout.component.ts:4]: Order of imports as below

import { Component } from '@angular/core';

import { Observable } from 'rxjs';

import { User } from '@iote/bricks';
import { UserService, AuthService} from '@ngfi/angular';

File: libs/elements/layout/user-menu/src/lib/components/app-logout/app-logout.component.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+@Component({
+  selector: 'app-app-logout',
+  templateUrl: './app-logout.component.html',
+  styleUrls: ['./app-logout.component.scss'],
+})
+export class AppLogoutComponent  {
+  user$: Observable<User>;
+
+  constructor(userService: UserService<User>, private _authService: AuthService) {}
```
Comment (IanOdhiambo9) [inline libs/elements/layout/user-menu/src/lib/components/app-logout/app-logout.component.ts:14]: move _authService below UserService

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html [hunk 11]
```diff
@@ -1,14 +1,16 @@
...
-<div *ngIf="user$ | async as user">
+<div *ngIf="user$ | async as user" class="outer">
   <button class="user-btn" mat-button [matMenuTriggerFor]="userMenu">
-    {{ user.photoUrl }}
-    <i class="far fa-user-circle user-image"></i> {{ user.displayName }}&nbsp;<!-- -&nbsp;{{user.phone}}&nbsp; -->
-    <i class="fas fa-caret-down"></i>
+    <div class="items">
+      <img src="{{ user.photoUrl }}" alt="" class="user-img">
+      <span class="username">{{ user.displayName }}.&nbsp;</span>
+      <fa-icon id="dropdown-icon" [icon]="dropdownIcon"></fa-icon>
```
Comment (IanOdhiambo9) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html:6]: Use CDN icons with <i> tag

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html [hunk 12]
```diff
@@ -1,14 +1,16 @@
...
+    </div>
+    <div class="role">
+      Admin&nbsp;
+    </div>
   </button>
-
-  <mat-menu #userMenu="matMenu" xPosition="before">
-    <button mat-menu-item>My profile</button>
-    <button mat-menu-item (click)="logout()">Log out</button>
```
Comment (IanOdhiambo9) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.html:11]: Use multilang for this text

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss [hunk 13]
```diff
@@ -1,5 +1,47 @@
...
+  margin-right: -0.6pc;
+  margin-bottom: 0.7pc;
+  padding-top: 0.5pc;
+}
+
+.outer:hover {
+  background-color: mat-color($accent, A1000);
+}
+
+#dropdown-icon {
```
Comment (IanOdhiambo9) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.scss:27]: use variable colors -> can be found in libs/elements/theming

eg. color: var(--convs-mgr-color-primary-yellow);

File: libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.ts [hunk 14]
```diff
@@ -1,31 +1,23 @@
...
+export class UserMenuComponent {
   user$: Observable<User>;
 
-  constructor(userService: UserService<User>,
-              private _router: Router,
-              private _authService: AuthService)
-  {
+  dropdownIcon = faCaretDown;
+
+  constructor(userService: UserService<User>, private _router: Router) {
```
Comment (IanOdhiambo9) [inline libs/elements/layout/user-menu/src/lib/components/user-menu/user-menu.component.ts:18]: move UserService below router

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**